src/events/antiSpam/messageCreate.ts

```diff
@@ -0,0 +1,77 @@
+import { Client, Constants, Message } from 'discord.js';
+import { inject, injectable } from 'tsyringe';
+import type { Event } from '../../Event';
+import { kRedis } from '../../tokens';
+import { on } from 'node:events';
+import { createHash } from 'node:crypto';
+import type { Redis } from 'ioredis';
+import { transformHashset } from '../../util/redisData';
+import { logger } from '../../logger';
+import { CaseAction, createCase } from '../../functions/cases/createCase';
+import { upsertCaseLog } from '../../functions/logs/upsertCaseLog';
+import { getGuildSetting, SettingsKeys } from '../../functions/settings/getGuildSetting';
+import i18next from 'i18next';
+import { SPAM_EXPIRE_SECONDS, SPAM_THRESHOLD } from '../../Constants';
+
+@injectable()
+export default class implements Event {
+	public name = 'Spam check';
+
+	public event = Constants.Events.MESSAGE_CREATE;
+
+	public constructor(public readonly client: Client<true>, @inject(kRedis) public readonly redis: Redis) {}
+
+	public async execute(): Promise<void> {
+		for await (const [message] of on(this.client, this.event) as AsyncIterableIterator<[Message]>) {
+			try {
+				if (!message.guild || !message.content.length) continue;
+
+				const contentHash = createHash('md5').update(message.content).digest('hex');
```

Comment:
We may want to transition this to a fuzzy hashing solution at some point in order to be somewhat more intelligent about spam bots that slightly vary their content to avoid automated filters like this.

Reply:
I will put a TODO here for now, as I'm lacking the theoretical knowledge for such an implementation at the moment (if you do happen to have resource recommendations on the topic I'd be grateful)